lib/ecto/adapters/mysql/connection.ex

```diff
@@ -514,7 +514,7 @@ if Code.ensure_loaded?(Mariaex.Connection) do
 
     defp ecto_to_db(:string),     do: "varchar"
     defp ecto_to_db(:float),      do: "double"
-    defp ecto_to_db(:datetime),   do: "datetime"
+    defp ecto_to_db(:datetime),   do: "datetime(6)"
```

Comment:
I think it's worth supporting older MySQL versions, because not all applications would be brand new. But I'm not sure about what the Ecto team think about it.
cc/ @josevalim @ericmj